### DISCUSSION TITLE
openclash+adguardad home时docker容器内dns解析域名出现问题

### DISCUSSION THREAD

**Discussion by 1053960054** [vernesong/OpenClash#2764] [Q&A]
我最开始是使用 openclash+adguardad home +Dnsmasq 
openclash设置模式为fake-ip ，dns设置取消本地DNS劫持，自定义上游dns服务器。
adguardad home  设置端口54，作为Dnsmasq 上游服务器，上游 DNS 服务器为127.0.0.1:7874
Dnsmasq 端口53，设置DNS 转发 127.0.0.1#54其它默认，
防火墙自定义规则就这两行iptables -t nat -A PREROUTING -p udp --dport 53 -j REDIRECT --to-ports 54
                                            iptables -t nat -A PREROUTING -p tcp --dport 53 -j REDIRECT --to-ports 54
![image](https://user-images.githubusercontent.com/38828427/197388816-38f13f7a-4353-453a-af2d-bc470c7c414e.png)
![image](https://user-images.githubusercontent.com/38828427/197388999-f13abffc-d0a7-4568-9e23-76907500b316.png)
![image](https://user-images.githubusercontent.com/38828427/197389017-69865ac3-2a4e-4a03-b31d-57ecc4917b7b.png)
![image](https://user-images.githubusercontent.com/38828427/197389028-56fe5310-9093-4481-9422-63b2a3a23dc2.png)
![image](https://user-images.githubusercontent.com/38828427/197389035-3239e635-dba8-4412-b4fa-c4a3a80efb9d.png)
![image](https://user-images.githubusercontent.com/38828427/197389048-5ad58978-0612-4645-b660-01b8b33668ef.png)
![image](https://user-images.githubusercontent.com/38828427/197389063-9ffefc9f-9d65-452d-90ca-ca69e3793637.png)
![image](https://user-images.githubusercontent.com/38828427/197389296-1a4fac0d-c530-4cf3-a539-b1ab240dd080.png)

这个时候
![image](https://user-images.githubusercontent.com/38828427/197389096-62f7de2d-8fa4-40d3-b824-2ed91c174e1f.png)
会出现不同的问题
1：可以正常科学上网，去广告也正常（但日志中出现大量ingest.sentry.io拦截），但是docker容器无法正常解析域名
![image](https://user-images.githubusercontent.com/38828427/197389232-757a2dcc-771d-4925-bc65-0c470cb16d85.png)
2：ip111前两项可以正确显示，但第三项从谷歌测试网络错误，这个时候docker容器内dns解析正常

![image](https://user-images.githubusercontent.com/38828427/197389192-5d76e395-735b-4ee3-9da0-c1f1d6d9917a.png)

然后我就说就openclash+adguardad home ，把Dnsmasq 端口改成其它的，去掉Dnsmasq ，adg的端口改为53，并且选择使用53端口替换Dnsmasq 
但还是会出现一样的问题，改来改去还是没弄好，

**Comment by 1053960054** [comment]
我自己感觉是dns的问题，但我一直没找到该怎么修改

**Comment by vernesong** [comment]
docker创建的时候要指定下DNS为本机
